File: practice/src/main/java/com/decade/practice/security/jwt/JwtUser.java
package com.decade.practice.security.jwt;

import com.decade.practice.entities.domain.entity.User;
import com.decade.practice.security.model.UserClaims;
import org.springframework.security.core.AuthenticatedPrincipal;

import java.io.Serializable;
import java.util.Objects;
import java.util.UUID;

public class JwtUser implements AuthenticatedPrincipal, Serializable {

      private UUID id;
      private String username;

      public JwtUser(UUID id, String username) {
            this.id = id;
            this.username = username;
      }

      protected JwtUser() {
      }

      public JwtUser(User user) {
            this(user.getId(), user.getUsername());
      }

      public void setId(UUID id) {
            this.id = id;
      }

      public String getUsername() {
            return username;
      }

      public void setUsername(String username) {
            this.username = username;
      }

      public JwtUser(UserClaims userClaims) {
            this(userClaims.getId(), userClaims.getUsername());
      }

      public UUID getId() {
            return id;
      }

      @Override
      public String getName() {
            return username;
      }

      @Override
      public int hashCode() {
            return Objects.hash(username);
      }

      @Override
      public boolean equals(Object other) {
            if (other == null) return false;
            return other.hashCode() == hashCode();
      }
}

File: practice/src/main/java/com/decade/practice/security/jwt/JwtUser.kt
package com.decade.practice.security.jwt

import com.decade.practice.model.domain.entity.User
import com.decade.practice.security.model.UserClaims
import org.springframework.security.core.AuthenticatedPrincipal
import java.io.Serializable
import java.util.*

class JwtUser(
      val id: UUID,
      val _username: String,
) : AuthenticatedPrincipal, Serializable {

      constructor(user: User) : this(
            user.id,
            user.username,
      )

      constructor(userClaims: UserClaims) : this(
            userClaims.id,
            userClaims.username,
      )

      override fun getName(): String = _username
      override fun hashCode(): Int = Objects.hash(_username)


      override fun equals(other: Any?): Boolean {
            return if (other == null) false
            else other.hashCode() == hashCode()
      }
}
